g_docs.go

```diff
@@ -904,3 +822,327 @@ func urlReplace(src string) string {
 	}
 	return strings.Join(pt, "/")
 }
+
+// constructObjectPropertie constructs a swagger.Propertie out of
+// an ast.Expr. This function recursively traverse all expression
+// until it reaches one of an object, selector, or pre-defined basic golang type.
+func constructObjectPropertie(field ast.Expr, packageName string, realTypes *[]string, pathInfo map[string]string) (propertie swagger.Propertie) {
+
+	// basic Go type can be directly translated into swagger type
+	// with pre-defined mapping
+	if basicType, ok := basicTypes[fmt.Sprint(field)]; ok {
+		propInfo := strings.Split(basicType, ":")
+
+		if len(propInfo) != 2 {
+			ColorLog("[ERRO] basicTypes const is not properly configured for %v", field)
+			os.Exit(1)
+		}
+
+		propertie.Type = propInfo[0]
+		propertie.Format = propInfo[1]
+
+		return
```

Comment:
One comment on [naked returns](https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters).

> Naked returns are okay if the function is a handful of lines. Once it's a medium sized function, be explicit with your return values. 

In general (but we dont have a clear guideline on this) we prefer no naked returns at all. We make exception tho as the example given in the link: when there are multiple return parameters of the same type.

This also applies to all other functions/methods here.